nextstep-di/src/main/java/nextstep/di/domain/context/GenericApplicationContext.java

```diff
@@ -0,0 +1,56 @@
+package nextstep.di.domain.context;
+
+import nextstep.annotation.ComponentScan;
+import nextstep.annotation.Configuration;
+import nextstep.di.domain.factory.BeanFactory;
+import nextstep.di.domain.factory.GenericBeanFactory;
+import nextstep.di.domain.scanner.ClassPathScanner;
+import nextstep.di.domain.scanner.ConfigurationScanner;
+import nextstep.di.support.SupportedClass;
+
+import java.lang.annotation.Annotation;
+import java.util.Set;
+
+public class GenericApplicationContext implements ApplicationContext {
+    private Class<?> configuration;
+    private BeanFactory beanFactory;
+    private SupportedClass supportedClass;
+
+    public GenericApplicationContext(Class<?> configuration) {
+        this.configuration = configuration;
+        this.beanFactory = new GenericBeanFactory();
+        this.supportedClass = new SupportedClass();
+    }
+
+    public GenericApplicationContext(Class<?> configuration, BeanFactory beanFactory) {
+        this.configuration = configuration;
+        this.beanFactory = beanFactory;
+        this.supportedClass = new SupportedClass();
+    }
+
+    @Override
+    public void initialize() {
+        ConfigurationScanner configurationScanner = new ConfigurationScanner(beanFactory);
+
+        if (configuration.isAnnotationPresent(ComponentScan.class)) {
+            ComponentScan componentScan = configuration.getAnnotation(ComponentScan.class);
+            ClassPathScanner classPathScanner = new ClassPathScanner(componentScan.value());
+            classPathScanner.scan(beanFactory);
+            classPathScanner.scan(supportedClass);
+        }
+
+        if (configuration.isAnnotationPresent(Configuration.class)) {
+            configurationScanner.register(configuration);
```

Comment:
`ConfigurationScanner configurationScanner = new ConfigurationScanner(beanFactory);` 또한 여기 내부에서 생성되어도 되겠네요. :) 

```suggestion
            ConfigurationScanner configurationScanner = new ConfigurationScanner(beanFactory);
            configurationScanner.register(configuration);
```